src/istio/chart/templates/tls-validation-job.yaml

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024 Defense Unicorns
+# SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: "{{ .Release.Name }}-tls-validation-sa"
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-install, pre-upgrade
+    "helm.sh/hook-weight": "-6"
+    "helm.sh/hook-delete-policy": hook-succeeded,hook-failed
```

Comment:
Open to discussion but I'd lean towards `hook-succeeded,before-hook-creation` here so that the job output is available on failure. Otherwise I think the job output would just disappear and make it hard to see what happened I think?